components/camel-xchange/src/main/java/org/apache/camel/component/xchange/XChangeMarketDataProducer.java

```diff
@@ -45,17 +40,15 @@ public XChangeEndpoint getEndpoint() {
     }
 
     @Override
-    protected int poll() throws Exception {
-        CurrencyPair pair = getEndpoint().getConfiguration().getCurrencyPair();
-        LOG.info("Going to execute ticker query for {}", pair);
-        
-        Ticker ticker = marketService.getTicker(pair);
-        
-        Exchange exchange = getEndpoint().createExchange();
-        exchange.getIn().setBody(ticker);
-        getProcessor().process(exchange);
+    public void process(Exchange exchange) throws Exception {
+
+        XChangeEndpoint endpoint = getEndpoint();
+        String method = endpoint.getConfiguration().getMethod();
         
-        return 1;
+        if (METHOD_TICKER.equals(method)) {
+            CurrencyPair pair = exchange.getIn().getHeader(HEADER_CURRENCY_PAIR, CurrencyPair.class);
+            Ticker ticker = marketService.getTicker(pair);
+            exchange.getOut().setBody(ticker);
```

Comment:
Dont set on OUT like that as it losses headers and whatnot during routing.  See this FAQ: http://camel.apache.org/using-getin-or-getout-methods-on-exchange.html

And there is a new getMessage() you can use btw

Reply:
Ok, got it.